stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2714,12 +2723,22 @@ impl StacksMessageCodec for NakamotoBlock {
     }
 
     fn consensus_deserialize<R: std::io::Read>(fd: &mut R) -> Result<Self, CodecError> {
-        let (header, txs) = {
+        panic!("NakamotoBlock should be deserialized with consensus_deserialize_with_epoch instead")
```

Comment:
I understand this function exists here because you need to `impl StacksMessageCodec` to get `consensus_serialize()`, but maybe we can separate `consensus_serialize()` and `consensus_deserialize()` into their own traits, and then have:

```rust
pub trait StacksMessageCodec: Serialize + Deserialize {}
pub trait StacksMessageCodecWithEpoch: Serialize + DeserializeWithEpoch {}
```

There will be other structs implementing `DeserializeWithEpoch`, so it's worth having to not have a bunch of potential panics in the code.

This will require replacing a lot of references to `StacksMessageCodec` with the more specific traits. Maybe this is a task for future PR...

Reply:
I'd drop this change.  You don't need `DeserializeWithEpoch` for `NakamotoBlock` since `NakamotoBlock` supports all transaction types anyway.